pkg/network/common/egressip.go

```diff
@@ -493,26 +494,13 @@ func (eit *EgressIPTracker) SetNodeOffline(nodeIP string, offline bool) {
 	eit.syncEgressIPs()
 }
 
-func (eit *EgressIPTracker) lookupNodeIP(ip string) string {
-	eit.Lock()
-	defer eit.Unlock()
-
-	if node := eit.nodesByNodeIP[ip]; node != nil {
-		return node.sdnIP
-	}
-	return ip
-}
-
 // Ping a node and return whether or not we think it is online. We do this by trying to
 // open a TCP connection to the "discard" service (port 9); if the node is offline, the
 // attempt will either time out with no response, or else return "no route to host" (and
 // we will return false). If the node is online then we presumably will get a "connection
 // refused" error; but the code below assumes that anything other than timeout or "no
 // route" indicates that the node is online.
 func (eit *EgressIPTracker) Ping(ip string, timeout time.Duration) bool {
-	// If the caller used a public node IP, replace it with the SDN IP
-	ip = eit.lookupNodeIP(ip)
```

Comment:
should clarify in the doc comment that you must pass an IP on the SDN, not an IP on the host network. Maybe also change the parameter name from `ip` to `sdnIP`?